src/query/expression/src/evaluator.rs

```diff
@@ -60,6 +62,8 @@ pub struct Evaluator<'a> {
     input_columns: &'a DataBlock,
     func_ctx: &'a FunctionContext,
     fn_registry: &'a FunctionRegistry,
+
+    cached_values: Mutex<HashMap<Expr, Value<AnyType>>>,
```

Comment:
better and a flag to use the `cached_values`.

cache_enable: bool

And only enable it in `block_operator`